File: 프로그래머스/l2_위장.py
from collections import defaultdict

def solution(clothes):
    answer = 1

    cloth_list = defaultdict(list)
    for val, key in clothes:
        cloth_list[key].append(val)

    for key in cloth_list:
        answer *= (len(cloth_list[key])+1)
    
    return answer-1


solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]])
solution([["crowmask", "face"], ["bluesunglasses", "face"], ["smoky_makeup", "face"]])

File: 프로그래머스/JAVA/l2_위장.java
import java.util.HashMap;
import java.util.Iterator;

public class l2_위장 {
    public static void main(String[] args){
        String[][] clothes = {{"yellowhat", "headgear"}, {"bluesunglasses", "eyewear"}, {"green_turban", "headgear"}};
        int answer = solution(clothes);

    }
    public static int solution(String[][] clothes) {
        int answer = 1;

        HashMap<String, Integer> clothMap = new HashMap<>();

        for (String[] cloth : clothes){
            System.out.println(cloth[0]+" "+cloth[1]);

            clothMap.put(cloth[1], clothMap.getOrDefault(cloth[1],0)+1);
            System.out.println(clothMap);
        }
        Iterator<Integer> it = clothMap.values().iterator();

        while(it.hasNext()){
            answer *= it.next().intValue()+1;
        }
        return answer-1;
    }
}
